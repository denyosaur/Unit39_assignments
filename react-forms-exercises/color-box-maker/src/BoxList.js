import React, { useState } from "react";
import Box from "./Box"
import NewBoxForm from "./NewBoxForm"
import uuid from 'react-uuid'

const BoxList = () => {
    const [boxes, setBoxes] = useState([{
        width: 100,
        height: 100,
        backgroundColor: "black",
        id: uuid()
    }]);
    const createBox = (height, width, backgroundColor) => {
        setBoxes(
            [...boxes, { height, width, backgroundColor, id: uuid() }]
        )
    };
    const deleteBox = (key) => {
        setBoxes(
            boxes.filter(box => {
                return box.id !== key
            })
        )
    };

    return <div className="BoxList">
        <div className="BoxList-form">
            <NewBoxForm createBox={createBox} />
        </div>
        <div className="BoxList-boxes">
            {
                boxes.map(box => {
                    return <Box key={box.id} height={box.height} width={box.height} backgroundColor={box.backgroundColor} deleteBox={() => deleteBox(box.id)} />
                })}
        </div>

    </div>

}

export default BoxList;