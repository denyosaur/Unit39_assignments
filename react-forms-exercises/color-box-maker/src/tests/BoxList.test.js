import { render, fireEvent } from '@testing-library/react';
import BoxList from '../BoxList';

//smoke test for Box
test('renders Box without crashing', () => {
    render(<BoxList />);
});

//snapshot test for Box
test('should match snapshot', () => {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

//render form and test box
test("render form", () => {
    const { getByText } = render(<BoxList />);
    const height = getByText("Height:");
    const width = getByText("Width:");
    const color = getByText("Color:");

    expect(height).toBeInTheDocument();
    expect(width).toBeInTheDocument();
    expect(color).toBeInTheDocument();
});

test("render test box", () => {
    const { getByText } = render(<BoxList />);
    const btn = getByText("X");

    expect(btn).toBeInTheDocument();
});

