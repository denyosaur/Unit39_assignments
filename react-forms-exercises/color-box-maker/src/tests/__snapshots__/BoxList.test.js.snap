// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <div
      class="BoxList-form"
    >
      <form>
        <label
          for="height"
        >
          Height:
        </label>
        <input
          name="height"
          placeholder="Enter Height Number"
          type="number"
          value=""
        />
        <label
          for="width"
        >
          Width: 
        </label>
        <input
          name="width"
          placeholder="Enter Width Number"
          type="number"
          value=""
        />
        <label
          for="color"
        >
          Color: 
        </label>
        <input
          name="color"
          placeholder="Enter Color"
          type="color"
          value="#000000"
        />
        <button>
          Create
        </button>
      </form>
    </div>
    <div
      class="BoxList-boxes"
    >
      <div
        style="background-color: black; height: 100px; width: 100px;"
      >
        <button>
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
